### PR TITLE
6.2 task solution

### PR DESCRIPTION
**Pull request by KovalAlexey**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,36 @@ public Integer getLength() {
...
+        Rectangle rectangle = (Rectangle) object;
+        if ((width == null && rectangle.width == null)
+                && (length == null && rectangle.length == null)
+                && (color == null && rectangle.color == null)) {
+            return true;
+        }
+        if ((width == rectangle.width) && (length == rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
+            return true;
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: let's combine these 
conditions

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,36 @@ public Integer getLength() {
...
+        return true;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: if no conditions gave you true , equals() still will return true?

Comment (KovalAlexey) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (KovalAlexey) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,35 @@ public Integer getLength() {
...
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if (width == null || length == null || color == null) {
+            return rectangle.width == null
+                    || rectangle.length == null
+                    || rectangle.color == null;
+        } else {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: how does if else blocks work?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,35 @@ public Integer getLength() {
...
+            return (width == rectangle.width)
+                    || (length == rectangle.length)
+                    || (color.equals(rectangle.color));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: let's combine these two conditions

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,34 @@ public Integer getLength() {
...
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        return (width == null || length == null || color == null)
+                ? rectangle.width == null
+                    || rectangle.length == null
+                    || rectangle.color == null
+                        : (width == rectangle.width)
+                        || (length == rectangle.length)
+                        || (color.equals(rectangle.color));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: thats complicated and hard to read. Let's  check each pair separately one by one

File: src/main/java/core/basesyntax/Rectangle.java [hunk 6]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || !object.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        return (width == null ? rectangle.width == null : width == rectangle.width)
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: Integer a = 128;
        Integer b = 128;
       boolean result = a == b; 
result will be true or false? and why(answer in comment below)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**